### PR TITLE
Igmp membership report packet serialization

### PR DESCRIPTION
**Pull request by ekarjun**
Adding support for serializing igmp v2 and v3 membership report packets.

### CODE REVIEW

**Comment by ekarjun** [comment]
This submission is related to serialization feature for igmp v2 and v3 membership report. Please look into the submission and let me know your comments.
@gconnell

**Comment by ekarjun** [comment]
@gconnell Would you please look into this submission and let me know your comments on it. I was not able to add reviewer when I created this pull request.

File: layers/igmp.go [hunk 1]
```diff
@@ -353,3 +353,48 @@ func decodeIGMP(data []byte, p gopacket.PacketBuilder) error {
 
 	return errors.New("Unable to determine IGMP type.")
 }
+
+// SerializeTo writes the serialized form of IGMP V2 and V3 packet layer into the
+// SerializationBuffer, implementing gopacket.SerializableLayer.
+func (igmp IGMP) SerializeTo(b gopacket.SerializeBuffer, opts gopacket.SerializeOptions) error {
+       data, err := b.PrependBytes(8915)
```
Comment (gconnell) [inline layers/igmp.go]: this seems really high... are all IGMP packets 8915 bytes?

From https://tools.ietf.org/html/rfc2236 and from https://tools.ietf.org/html/rfc3376 I'd expect these packets to be far smaller.

Bytes should only be prepended/appended to 'b' if they're going to be part of the final packet.

So, given the code below, I'd expect PrependBytes(8) for v2 and PrependBytes(8 + x*#groups) or something like that.

Comment (ekarjun) [reply layers/igmp.go]: I'll implement a logic to calculate it with respect to the igmp version of the packet received and submit this again.

Comment (ekarjun) [reply layers/igmp.go]: Hi @gconnell,
I have submitted a new commit for the fix. Please go through the code and let me know if its good. In byte size calculation for igmpv3 packets, I haven't considered Auxiliary data as its not being used according to RFC 3376.

**Comment by ekarjun** [comment]
Hi @gconnell, would you please take look at my latest commit and see if its good for merge?
Regards,
Arjun

**Pull request closed without merging**